decode/layout.ts

```diff
@@ -1,58 +1,60 @@
 import hash from "../src/data/hash";
 import { resolve } from "../src/data/token";
-import { Event, IDecodedEvent, Token } from "../types/data";
-import { IAttributes, IBoxModel, IChecksum, IDecodedNode, IDocumentSize, ILayout, IResource } from "../types/layout";
+import { Event, Token } from "../types/data";
+import { ChecksumEvent, DomData, LayoutEvent, ResourceEvent } from "../types/decode";
+import { Attributes, BoxModelData, ChecksumData, DocumentData, ResourceData } from "../types/layout";
 
 const ID_ATTRIBUTE = "data-clarity";
 let placeholderImage = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR4nGNiOAMAANUAz5n+TlUAAAAASUVORK5CYII=";
-let selectors: ILayout[];
-let resources: IResource[];
+export let checksums: ChecksumData[];
+export let resources: ResourceData[];
 let lastTime: number;
 
 export function reset(): void {
-    selectors = [];
+    checksums = [];
     resources = [];
     lastTime = null;
 }
 
-export function decode(tokens: Token[]): IDecodedEvent {
+export function decode(tokens: Token[]): LayoutEvent {
     let time = lastTime = tokens[0] as number;
     let event = tokens[1] as Event;
-    let decoded: IDecodedEvent = {time, event, data: []};
 
     switch (event) {
         case Event.Document:
```

Comment:
>Document [](start = 19, length = 8)

I saw lot of examples of this event with the same values repeated for the same page (i.e. identical width/height). When do we send it?

Reply:
Already fixed the repetitions. No longer an issue.